security/subsystem/src/main/java/org/jboss/as/security/SecurityDomainAdd.java

```diff
@@ -182,6 +183,10 @@ public void launchServices(OperationContext context, String securityDomain, Mode
                 .addDependency(JaasConfigurationService.SERVICE_NAME, Configuration.class,
                         securityDomainService.getConfigurationInjector());
 
+        if (jsseSecurityDomain != null) {
+            builder.addDependency(ContextNames.JBOSS_CONTEXT_SERVICE_NAME.append("jaas"));
```

Comment:
Although this may correct the ordering for this bug report this feels a little odd - is there a reason within the Security Subsystem that a security domain should depend on a specific naming context?  And if so why would that be exclusive to if JSSE is defined?

Reply:
The error scenario was:
If IIOP subsystem uses legacy security domain then ORB service has dependency on it.
When security domain starts it registers it's context within JNDIBasedSecurityManagement.
The problem is that the BinderService which is responsible for binding the contexts within JNDI is independent of security domain.
As a result following race condition was happening: security domain starts -> orb starts and ask for jsse domain -> failure (binder service hasn't registered the context yet)
I thought that it would be better to add dependency between security domain and binder that directly from IIOP because security domain is unreachable by JNDI before Binder has started. 
If this solution is acceptable, than I will look indeed whether there are others types of domains that need to wait for their context being registered. If not, then we need to find another way to solve it.